x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/job/RollupJobStats.java

```diff
@@ -5,152 +5,27 @@
  */
 package org.elasticsearch.xpack.core.rollup.job;
 
-import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Writeable;
-import org.elasticsearch.common.xcontent.ConstructingObjectParser;
-import org.elasticsearch.common.xcontent.ToXContentObject;
-import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.indexing.IndexerStats;
 
 import java.io.IOException;
-import java.util.Objects;
-
-import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
 
 /**
- * This class holds the runtime statistics of a job.  The stats are not used by any internal process
- * and are only for external monitoring/reference.  Statistics are not persisted with the job, so if the
- * allocated task is shutdown/restarted on a different node all the stats will reset.
+ * This class holds the runtime statistics of a rollup job, derived from {@link IndexerStats}}.
+ *
  */
-public class RollupJobStats implements ToXContentObject, Writeable {
-
-    public static final ParseField NAME = new ParseField("job_stats");
-
-    private static ParseField NUM_PAGES = new ParseField("pages_processed");
-    private static ParseField NUM_DOCUMENTS = new ParseField("documents_processed");
-    private static ParseField NUM_ROLLUPS = new ParseField("rollups_indexed");
-    private static ParseField NUM_INVOCATIONS = new ParseField("trigger_count");
-
-    private long numPages = 0;
-    private long numDocuments = 0;
-    private long numRollups = 0;
-    private long numInvocations = 0;
-
-    public static final ConstructingObjectParser<RollupJobStats, Void> PARSER =
-            new ConstructingObjectParser<>(NAME.getPreferredName(),
-                    args -> new RollupJobStats((long) args[0], (long) args[1], (long) args[2], (long) args[3]));
+public class RollupJobStats extends IndexerStats {
```

Comment:
This class is empty as there is nothing in there specific to rollup right now, but that might change.

Note: I changed field names! So this break BWC for the get rollup jobs api. Seeking advise, whether this is a problem and whether I should implement a compat layer here.

Reply:
I think we should use the deprecated names feature of `ParseField` so we avoid a bwc break here. Its true that rollups is experimental so we can in theory make bwc breaks without notice but I think where its easy for us to deprecate we should 😄